### DISCUSSION TITLE
Feature Request HmIP-SWSD: Store last primary and last secondary alarm time

### DISCUSSION THREAD

**Discussion by LinusHoppe** [SukramJ/aiohomematic#751] [Ideas]
I have a lot of HmIP-SWSD in my environment. As they are (without CCU) connected and share alarms, it is not easy to find out which device has triggered the alarm.
You can distinguish between `primary` and `secondary` alarm and I am using that for automations and notifications. But as the alarm is quite loud my first aim is to stop the alarm by hitting the device button. After that, all sensors go back to `idle` and I cannot see which device has triggered the alarm.

It would be really helpful to have an attribute/sensor within every HmIP-SWSD device where the time of the last primary alarm and the time of the last secondary alarm is stored.

**Comment by SukramJ** [comment]
We don't cache states outside of the devices. 

You could create an automation by yourself, that stores the last device in a text helper.

**Comment by hklages** [comment]
A work around would be to use a ccu program, write the outcome in a system variable an read that in home assistant. 
[See this discussion ](https://homematic-forum.de/forum/viewtopic.php?f=26&t=29433&p=748810&hilit=rauchmelder#p748810)

Havent tried it - just an idea. 

I do similiar things to read the System Messages (low battery) and process them in Home Assistant - that works.
